### PR TITLE
Drop PG 9.5 support and warn if running with lower than GEOS 3.8

### PR DESCRIPTION
**Pull request by Algunenano**

### CODE REVIEW

File: postgis/sqldefines.h.in [hunk 1]
```diff
@@ -14,13 +14,8 @@
 #define POSTGIS_LIBXML2_VERSION @POSTGIS_LIBXML2_VERSION@
 #define POSTGIS_SFCGAL_VERSION @POSTGIS_SFCGAL_VERSION@
 
-#if POSTGIS_PGSQL_VERSION >= 96
 #define _PARALLEL PARALLEL SAFE
```
Comment (Komzpa) [inline postgis/sqldefines.h.in]: can we unroll this define please?

Comment (Algunenano) [reply postgis/sqldefines.h.in]: I was wondering whether to do it or not. Do you think it's worth changing all those lines?

Comment (pramsey) [reply postgis/sqldefines.h.in]: I don't know from "worth it"... I'm going to go with "yes". We used to have STRICT and friends in defines, and we took them out, and didn't suffer, so the one time big patch isn't a huge problem, in my mind.

Comment (Komzpa) [reply postgis/sqldefines.h.in]: Yes. Less copy paste during one-liner debug.

Comment (Algunenano) [reply postgis/sqldefines.h.in]: I'll do it then :wink:

Comment (Algunenano) [reply postgis/sqldefines.h.in]: `postgis_brin.sql.in` doesn't have PARALLEL tags, so I'll try to add them in a different commit.

**Pull request merged**